### ISSUE TITLE
Credential Provider Not Found at Default Path on Linux 

### ISSUE THREAD

**Issue by twinter-amosfivesix** [OneGet/NuGetProvider#68]
### TLDR:

[This line](https://github.com/OneGet/NuGetProvider/blob/master/NugetLightRequest.cs#L1985) in `NugetLightRequest.GetCredsFromCredProvider` incorrectly uses `$HOME` instead of `%HOME%` [as required](https://github.com/dotnet/runtime/issues/25792) by `Environment.ExpandEnvironmentVariable()`. This causes the credential provider to not be found

```
// If running Unix
path = "$HOME/.nuget/plugins/netcore/CredentialProvider.Microsoft/CredentialProvider.Microsoft.dll";
```

### Longer Version

Running Red Hat Enterprise Linux release 8.7, PowerShell 7.3.1, PackageManagement 1.4.8.1, PowerShellGet 2.2.5. 

Trying to use an [Azure Artifacts feed with PowerShell](https://learn.microsoft.com/en-us/azure/devops/artifacts/tutorials/private-powershell-library?view=azure-devops). Installed the credential provider, setup my PS-Repository, etc. (Got all this working on Windows no problem). 

When I run `Install-Module ModuleFromMyFeed -Debug` I get output that includes this:
```
DEBUG: 00:00:01.3022327 Calling 'GetCredsFromCredProvider' on https://pkgs.dev.azure.com:443/myorg/_packaging/myfeed/nuget/v2/FindPackagesById()?id='ModuleFromMyFeed'&$skip=%7B0%7D&$top=%7B1%7D
DEBUG: 00:00:01.3028816 Calling credential provider installed at
Could not execute because the specified command or file was not found.
Possible reasons for this include:
  * You misspelled a built-in dotnet command.
  * You intended to execute a .NET program, but dotnet--V does not exist.
  * You intended to run a global tool, but a dotnet-prefixed executable with this name could not be found on the PATH.
```

Note the line for `Calling credential provider installed at`. There is nothing listed after the `at`. And note how `dotnet` thinks I'm trying to run `dotnet --V`.

If you read through the code in [NugetLightRequest.GetCredsFromCredProvider](https://github.com/OneGet/NuGetProvider/blob/master/NugetLightRequest.cs#L1946-L2049) you'll see it first looks for `NUGET_PLUGIN_PATHS`, which I do **not** have defined. That fails so it tries to look for the provider at `"$HOME/.nuget/plugins/netcore/CredentialProvider.Microsoft/CredentialProvider.Microsoft.dll"`. It passes that string to `Environment.ExpandEnvironmentVariables()`. But does *not work*. Because `Environment.ExpandEnvironmentVariables` expects variables to be in the format `%VAR%`, not `$VAR`. See this [github issue](https://github.com/dotnet/runtime/issues/25792). So the variable is not expanded, so the `File.Exists()` check fails. That leaves `credProviderPath` blank, so the code then tries to run `dotnet  -V verbose -U <url>`. Which gives the error we see in the output.

### The Fix
In the source code switch `$HOME` to `%HOME%`.

### The workaround.

Set the environment variable `NUGET_PLUGIN_PATHS` to the full path of `CredentialProvider.Microsoft.dll`. For example, `$env:NUGET_PLUGIN_PATHS = "/home/username/.nuget/plugins/netcore/CredentialProvider.Microsoft/CredentialProvider.Microsoft.dll"`

This does work for me.

**Comment by jachin84** [comment]
I ran into this today as well and can confirm @twinter-amosfivesix's findings, fix and workaround. This was extremely difficult to find.

**Comment by et1975** [comment]
The year is 2025, PowerShell still reports obscure messages like "You intended to execute a .NET program, but dotnet--V does not exist."

**Comment by ThomasNieto** [comment]
@et1975 [OneGet](https://github.com/OneGet/oneget) is deprecated. You can try my replacement called [AnyPackage](https://github.com/anypackage/anypackage) which has a [NuGet](https://github.com/anypackage/nuget) provider.
